modules/ROOT/pages/installation/linux/debian.adoc

```diff
@@ -55,21 +55,58 @@ java-1.17.0-openjdk-amd64 1711 /usr/lib/jvm/java-1.17.0-openjdk-amd64
 java-1.11.0-openjdk-amd64 1071 /usr/lib/jvm/java-1.11.0-openjdk-amd64
 ----
 
-* Identify your Java 17 version, in this case, it is `java-1.17.0-openjdk-amd64`.
-Then set it as the default with (replacing `<java17name>` with the appropriate name from above):
+. Identify your Java 17 version:
 +
 [source, shell]
 ----
-sudo update-java-alternatives --jre --set <java17name>
+java -version
 ----
++
+In this case, it is `java-1.17.0-openjdk-amd64`.
 
-* Finally, confirm which version of Java is the default:
+. Set it as the default by replacing `<java17name>` with its name:
 +
 [source, shell]
 ----
-java -version
+sudo update-java-alternatives --jre --set <java17name>
+----
+
+. Confirm which version of Java is the default using `java -version` again.
+
+[[debian-install-newer-java]]
+=== If `apt` installed OpenJDK 19
+
+On newer Debian or Ubuntu operating systems, Java 19 is available by default, and `apt` may have installed OpenJDK 19, *even if Java 17 was already installed*.
+
+If this happens, you will see this warning on Neo4j start:
+[output]
+----
+WARNING! You are using an unsupported Java runtime.
+* Please use Oracle(R) Java(TM) 17, OpenJDK(TM) 17 to run Neo4j.
+* Please see https://neo4j.com/docs/ for Neo4j installation instructions.
+----
+
+To fix this, you can install Java 17 manually, uninstall OpenJDK 19, or set Java 17 as the default.
```

Comment:
Sadly no. If you uninstall java 19 then apt will helpfully also uninstall neo4j UNLESS there's a compatible java version already installed.
So basically what happens is that:
1. neo4j says it has a dependency on java-17. 
2. OpenJDK 19 package says that it provides a compatible java-17 package. 
3. Java 19 is a more recent version than java 17, so apt assumes that it's better.
4. apt """helpfully""" installs the more recent java 19 package.
5. we come along and uninstall java 19 because we don't want it.
6. nothing provides a java 17 any more, so neo4j has a missing dependency.
7. apt also deletes neo4j because it won't be able to run without a java 17.

To stop apt deleting neo4j in step 6, we need to have java 17 installed already so that there is a package installed which provides java-17.

I suppose there are really two fix options:
1. install java 17 and uninstall java 19
2. install java 17 and set it to default so that java 19 is ignored.